jaxrs/src/main/java/org/killbill/billing/jaxrs/resources/InvoiceResource.java

```diff
@@ -389,18 +389,8 @@ public Response generateDryRunInvoice(@Nullable final InvoiceDryRunJson dryRunSu
                                           @javax.ws.rs.core.Context final HttpServletRequest request,
                                           @javax.ws.rs.core.Context final UriInfo uriInfo) throws AccountApiException, InvoiceApiException {
         final CallContext callContext = context.createCallContextWithAccountId(accountId, createdBy, reason, comment, request);
-        final LocalDate inputDate;
-        if (dryRunSubscriptionSpec != null) {
-            if (DryRunType.UPCOMING_INVOICE.equals(dryRunSubscriptionSpec.getDryRunType())) {
-                inputDate = null;
-            } else if (DryRunType.SUBSCRIPTION_ACTION.equals(dryRunSubscriptionSpec.getDryRunType()) && dryRunSubscriptionSpec.getEffectiveDate() != null) {
-                inputDate = dryRunSubscriptionSpec.getEffectiveDate();
-            } else {
-                inputDate = toLocalDate(targetDate);
-            }
-        } else {
-            inputDate = toLocalDate(targetDate);
-        }
+        final LocalDate inputDate = (dryRunSubscriptionSpec != null && DryRunType.UPCOMING_INVOICE.equals(dryRunSubscriptionSpec.getDryRunType())) ?
+                                     null : toLocalDate(targetDate);
```

Comment:
Since `dryRunSubscriptionSpec.getEffectiveDate()` isn't used anymore for `SUBSCRIPTION_ACTION`, do we need to update the Slate docs to reflect this change?

Reply:
No because [it is still used](https://github.com/killbill/killbill/blob/master/jaxrs/src/main/java/org/killbill/billing/jaxrs/resources/InvoiceResource.java#L1076). This is the whole point, we need 2 distinct dates, one for the action effective date and one for the target invoice date.

Reply:
Actually, i do need to update the doc as it is wrong: `Target date, if dryRunType is TARGET_DATE. Otherwise not needed`. Will do...